### PR TITLE
33695 radio button constraint validation

### LINKED ISSUE #33695: Incorrect validation description
### MDN URL

https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/radio

### What specific section or headline is this issue about?

validation

### What information was incorrect, unhelpful, or incomplete?

It is said that

> Radio buttons don't participate in constraint validation; they have no real value to be constrained.

### What did you expect to see?

However, they do participate in at least `required` validation, here's screenshot from FireFox, but all other major browsers also implement this: 
image

### Do you have any supporting links, references, or citations?

_No response_

### Do you have anything more you want to share?

here's the code used to make the screenshot:
```html
      <input type="radio" required name="foo" value="bar" id="bar" /><label for="bar">bar</label>
      <input type="radio" required name="foo" value="kar" id="kar" /><label for="kar">kar</label>
      <input type="radio" required name="foo" value="jar" id="jar" /><label for="jar">jar</label>
      <input type="radio" required name="foo" value="dar" id="dar" /><label for="dar">dar</label><br />
```
So the name is the same and one or more radio buttons have `required` attribute.

### MDN metadata

<details>
<summary>Page report details</summary>

* Folder: `en-us/web/html/element/input/radio`
* MDN URL: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/radio
* GitHub URL: https://github.com/mdn/content/blob/main/files/en-us/web/html/element/input/radio/index.md
* Last commit: https://github.com/mdn/content/commit/72ca3d725e3e56b613de3ac9727bd0d6d619c38a
* Document last modified: 2024-01-08T13:33:20.000Z

</details>

### PR DESCRIPTION
**Pull request by estelle**
fixes #33695 33695

### CODE REVIEW

File: files/en-us/web/html/element/input/radio/index.md [hunk 1]
```diff
@@ -182,7 +182,7 @@ Beyond accessibility, this is another good reason to properly set up `<label>` e
 
 ## Validation
 
-Radio buttons don't participate in constraint validation; they have no real value to be constrained.
+While radio buttons have no real value to be constrained, if any radio button in a same-name group of radio buttons has the `required` attribute set, and none of the radio buttons is checked, the [`valueMissing`](/en-US/docs/Web/API/ValidityState/valueMissing) property of a {{domxref("ValidityState")}} object will return true during validation.
```
Comment (chrisdavidmills) [inline files/en-us/web/html/element/input/radio/index.md:185]: This sounds a little over-complicated to me. For example, any radio button with `required` set will trigger the specified validation, not just ones in a group, and the description of the validation will probably sound a bit confusing to some, without any context of what that is in real terms (i.e. the browser telling you that you need to select an option).

How about something like this:

```suggestion
While radio buttons have no real value to be constrained, they can be set as required. Any radio button with the [`required`](/en-US/docs/Web/HTML/Attributes/required) attribute set (either directly, or via another radio button in a same-name group) will trigger the browser's in-built validation if no option is selected when form submission is attempted. The browser will ask the user to select an option. This is equivalent to the [`valueMissing`](/en-US/docs/Web/API/ValidityState/valueMissing) property of a {{domxref("ValidityState")}} object returning true during validation.
```

Hrm, still seems a bit wordy. Let me know if that helps at all.

**Review by chrisdavidmills — CHANGES_REQUESTED** [review]
Some thoughts

File: files/en-us/web/html/element/input/radio/index.md (hunk 1 above)
Comment (estelle) [reply files/en-us/web/html/element/input/radio/index.md:185]: It is not the equivalent of valueMissing; it is ValueMissing . i'll rework it to include a single required.

Comment (estelle) [inline files/en-us/web/html/element/input/radio/index.md:185]:
```suggestion
If a radio button is required or a same-named group or radio buttons has at least one member with `required` set, and no radio buttons is checked, the [`valueMissing`](/en-US/docs/Web/API/ValidityState/valueMissing) property of a {{domxref("ValidityState")}} object will return true during validation.
```

Comment (chrisdavidmills) [reply files/en-us/web/html/element/input/radio/index.md:185]: Ah, OK, so it is actually what `required` uses under the hood! Cool, I didn't know that.

**Review by chrisdavidmills — APPROVED** [review]
I had another suggestion for you. I'll approve this time so you can put this one to bed.

File: files/en-us/web/html/element/input/radio/index.md (hunk 1 above)
Comment (estelle) [inline files/en-us/web/html/element/input/radio/index.md:185]:
```suggestion
In the case of a radio button with the [`required`](/en-US/docs/Web/HTML/Attributes/required) attribute set or a same-named group of radio buttons in which at least one member has required set, a radio button needs to be selected for the control to be considered valid. If no radio button is checked, the [`valueMissing`](/en-US/docs/Web/API/ValidityState/valueMissing) property of a {{domxref("ValidityState")}} object will return true during validation, and the browser will ask the user to select an option.
```

File: files/en-us/web/html/element/input/radio/index.md [hunk 2]
```diff
@@ -182,7 +182,7 @@ Beyond accessibility, this is another good reason to properly set up `<label>` e
 
 ## Validation
 
-Radio buttons don't participate in constraint validation; they have no real value to be constrained.
+In the case of a radio button with the [`required`](/en-US/docs/Web/HTML/Attributes/required) attribute set or a same-named group of radio buttons in which at least one member has required set, a radio button needs to be selected for the control to be considered valid. If no radio button is checked, the [`valueMissing`](/en-US/docs/Web/API/ValidityState/valueMissing) property of a {{domxref("ValidityState")}} object will return true during validation, and the browser will ask the user to select an option.
```
Comment (estelle) [inline files/en-us/web/html/element/input/radio/index.md:185]:
```suggestion
In the case of a radio button with the [`required`](/en-US/docs/Web/HTML/Attributes/required) attribute set, or a same-named group of radio buttons in which at least one member has `required` set, a radio button needs to be selected for the control to be considered valid. If no radio button is checked, the [`valueMissing`](/en-US/docs/Web/API/ValidityState/valueMissing) property of a {{domxref("ValidityState")}} object will return `true` during validation, and the browser will ask the user to select an option.
```
